myCryptoWallet-ConsoleApp/src/main/java/com/isa/CoinsList.java

```diff
@@ -0,0 +1,60 @@
+package com.isa;
+
+import java.util.*;
+
+public class CoinsList {
+
+    private Map<Integer, List<String>> pagesMap;
+    private List<String> coins;
+    private int totalPages;
+    private int recordsPerPage;
+
+    public CoinsList(int recordsPerPage) {
+        this.recordsPerPage = recordsPerPage;
+        this.coins = Endpoints.getEndpoints();
+        setTotalPages();
+        pagesCreator();
+        openPageFromKeyboard();
+    }
+
+    public void pagesCreator() {
+        this.pagesMap = new LinkedHashMap<>();
+
+        for (int i = 0; i < totalPages; i++) {
+            List<String> temporaryCoinList;
+            if (coins.size() < recordsPerPage) temporaryCoinList = coins.subList(0, coins.size());
+            else if (i == totalPages - 1) {
+                temporaryCoinList = coins.subList(i * recordsPerPage, coins.size());
+            } else temporaryCoinList = coins.subList(i * recordsPerPage, (i + 1) * recordsPerPage);
+
+            pagesMap.put(i, temporaryCoinList);
+        }
+    }
+
+    public List<String> openPage(Integer pageNumber){
+        System.out.println(pagesMap.get(pageNumber - 1));
+        return pagesMap.get(pageNumber - 1);
+    }
+
+    public void openPageFromKeyboard(){
+
+        int pageNumber = 0;
+        do {
+            System.out.println("podaj numer strony (total pages: " + totalPages + ")");
+            System.out.println("Aby wyjść podaj \"0\"");
+            Scanner scanner = new Scanner(System.in);
+            pageNumber = Integer.parseInt(scanner.nextLine());
```

Comment:
tu musisz wrzucić try{}catch żeby zrobić wyjątek jakby użytkownik wpisał przez przypadek jakieś inne znaki niż cyfry

Reply:
nie dodawałem try catch bo jak aplikacja bedzie już webowa to użytkownik będzie przełączał strony używając przycisków "next page" albo wybierał numer strony z listy a nie wpisywał numer strony z klawiatury. ale jak chcesz mogę dodać ;-)